### ISSUE TITLE
Inconsistent and unexpected intersection when using `TypeIs[Type[<something>]]`

### ISSUE THREAD

**Issue by Daraan** [microsoft/pyright#9002]
**Describe the bug**

In my expectation the following code should pass without problems, the guard should create an outer type intersection
`type[<subclass of T1 and T2>]` and recognizable as type, however here pyright infers the result as an instance and not a type.

Ignore that the types in this case are incompatible. A real situation is for example when using `is_dataclass` on a `type[T1]` that is not 100% inferable as a dataclass. This seems to be a general issue in some situation it works, however in some cases the (type & type) condition fails with TypeIs to produce a type again.

```python
from typing import *
from typing_extensions import TypeIs

# could be any most types
T1: TypeAlias = dict
T2: TypeAlias = int

value = cast(type[T1], dict)

def guard(x: Type[Any]) -> TypeIs[Type[T2]]:
    return bool(x)

def foo(x: Type[Any]):
    ...
   
assert guard(value)

reveal_type(value)
# Type of "value" is "<subclass of type[dict] and type[int]>"
# Expected a type[<subclass of ... and ...>]

foo(value)
# error: Argument of type "<subclass of type[dict] and type[int]>" 
# cannot be assigned to parameter "x" of type "Type[Any]" in function "foo"
# "<subclass of type[dict] and type[int]>" is not assignable to "Type[Any]"

```

**VS Code extension or command-line**
CLI: pyright 1.1.380

**Comment by erictraut** [comment]
Thanks for the bug report. I agree this is incorrect behavior.

Pyright's old behavior (prior to 1.1.380) was also incorrect in this case. It produced an intersection of `Never`. Mypy also produces `Never` in this case, so it has a similar bug.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.381.
